runelite-client/src/main/java/net/runelite/client/plugins/mining/Rock.java

```diff
@@ -46,52 +46,52 @@
 
 enum Rock
 {
-	TIN(Duration.ofMillis(2300), ROCKS_11360, ROCKS_11361),
-	COPPER(Duration.ofMillis(2200), ROCKS_11161),
-	IRON(Duration.ofMillis(5300), ROCKS_11364, ROCKS_11365)
+	TIN(Duration.ofMillis(2300), 0, ROCKS_11360, ROCKS_11361),
```

Comment:
Instead of adding `0` for each rock which doesn't need any offset, create a ctor overload which calls the new ctor with `0` height.

Reply:
Is this possible? Since the constructor takes `(Duration, int, int...)`, how would it know that the second parameter is missing? It would just treat the first `int` from what is meant to be the `int... ` part as the second parameter instead wouldn't it?

Reply:
Instead of removing the existing ctor, add a second one. Then they should look like this:

```java
Rock(Duration respawnTime, int... ids)
{
	this(respawnTime, 0, ids);
}
Rock(Duration respawnTime, int zOffset, int... ids)
{
	this.respawnTime = respawnTime;
	this.zOffset = zOffset;
	this.ids = ids;
}
```

You can see this behavior of defining multiple overloaded constructors in other classes like [AgilityShortcut](https://github.com/runelite/runelite/blob/c1b0ca4ffdf95e6e6979cdf2058b4522530f742a/runelite-client/src/main/java/net/runelite/client/game/AgilityShortcut.java#L230-L242)